server/src/main/java/org/eclipse/openvsx/RegistryAPI.java

```diff
@@ -388,6 +392,14 @@ private int mergeSearchResults(SearchResultJson result, List<SearchEntryJson> en
         if (Strings.isNullOrEmpty(namespace.name)) {
             return ResponseEntity.ok(ResultJson.error("Missing required property 'name'."));
         }
+        var principal = users.getOAuth2Principal();
+        UserData user = null;
+        if (principal != null) {
+            user = users.updateUser(principal);
+        }
+        if(token == null && (user == null || !user.getRole().equals("admin"))) {
+            return ResponseEntity.ok(ResultJson.error("Missing token or admin rights."));
+        }
```

Comment:
Authentication is already checked in `local.createNamespace(namespace, token)`, no need to do it twice.